x-pack/apm-server/aggregation/spanmetrics/aggregator.go

```diff
@@ -227,9 +227,6 @@ func (a *Aggregator) ProcessBatch(ctx context.Context, b *model.Batch) error {
 }
 
 func (a *Aggregator) processSpan(event *model.APMEvent) {
-	if event.Span.DestinationService == nil || event.Span.DestinationService.Resource == "" {
-		return
-	}
```

Comment:
```go
	if event.Service.Target == nil {
		return
	}
```

I think this should be updated to NOT process if service target is not set, provided there are no other edge cases. IIUC, there will be three cases when we should produce the service destination aggregation:

1. `service.target.*` is not set and `destination_service.resource` is set
2. `service.target.*` is set and `destination_service.resource` is not set
3. both are set

For `1`, when decoding we will infer `service.target.*` and set it whereas for `2` and `3` we will not infer anything ([ref](https://github.com/elastic/apm-data/blob/c0b8c4f469fdc878aec0ea0e488cb084baa66605/input/elasticapm/internal/modeldecoder/v2/decoder.go#L1032-L1035)).

Reply:
It can. But since today, there can be aggregations emitted if there is no service target (but there is a destination), that seemed like a strange change.

Reply:
> But since today, there can be aggregations emitted if there is no service target (but there is a destination), that seemed like a strange change

Is this possible, I think when we decode an event with no service target but destination, we would infer the service target: https://github.com/elastic/apm-data/blob/c0b8c4f469fdc878aec0ea0e488cb084baa66605/input/elasticapm/internal/modeldecoder/v2/decoder.go#L1032-L1035

Reply:
That makes sense, thanks. I've updated the PR.